src/browser/synthesis-ui.js

```diff
@@ -127,73 +123,101 @@
+
+  const updateBySelectedWebView = (state, action) =>
+    updateByWebViewIndex(state, state.webViews.selected, action);
+
   const updateByInputAction = (state, action) =>
     action instanceof Input.Submit ? submit(state, action.value) :
-    action instanceof Focusable.Focus ? editWebViewByID(state, null) :
-    action instanceof Focusable.Focused ? editWebViewByID(state, null) :
+    action instanceof Focusable.Focus ? editSelectedWebView(state) :
+    action instanceof Focusable.Focused ? editSelectedWebView(state) :
     state;
 
-  const completeSelection = state =>
+  const fadeToShowModeFromSelectMode = state =>
     state.mode === 'select-web-view' ? fadeToShowMode(state) :
     state;
 
+  const activateSelectedWebView = state =>
+    state.update('webViews', WebView.activate);
+
+  const completeSelection = compose(
+    fadeToShowModeFromSelectMode,
+    activateSelectedWebView);
+
   const escape = state =>
     // If we're already showing a webview, or we can't show a webview because
     // none exist yet, do nothing. Otherwise, fade to the selected web view.
     state.mode === 'show-web-view' || state.webViews.selected === null ? state :
-    fadeToWebViewByID(state);
+    fadeToShowMode(state);
 
   const update = (state, action) =>
     action instanceof Navigation.Stop ?
       escape(state) :
-    action instanceof WebView.Action ?
-      updateByWebViewAction(state, action.id, action.action) :
-    action instanceof WebView.Open ?
-      updateByWebViewAction(state, null, action) :
-    action instanceof WebView.FadeToOpen ?
-      updateByWebViewAction(state, null, action) :
-    action instanceof WebView.Close ?
-      updateByWebViewAction(state, null, action) :
-    action instanceof WebView.SelectNext ?
-      updateByWebViewAction(state, null, action) :
-    action instanceof WebView.SelectPrevious ?
-      updateByWebViewAction(state, null, action) :
     action instanceof Input.Action ?
       updateByInputAction(state, action.action) :
     action instanceof Input.Submit ?
       updateByInputAction(state, action) :
+    action instanceof Preview.Create ?
+      createWebView(state, 'zoom') :
+    action instanceof OpenNew ?
+      createWebView(state, 'fade') :
+
+    action instanceof WebView.ByID ?
+      updateByWebViewID(state, action.id, action.action) :
+    action instanceof WebView.BySelected ?
+      updateBySelectedWebView(state, action.action) :
+
     action instanceof Gesture.Pinch ?
-      showPreview(state) :
-    action instanceof Select ?
+      zoomEditSelectedWebView(state) :
+    action instanceof ShowSelected ?
       completeSelection(state) :
+    action instanceof ShowPreview ?
+      showPreview(state) :
     state;
-
   exports.update = update;
+
+
+  const service = address => {
+    let id = -1;
+    const showPreview = address.pass(ShowPreview)
+
+    return action => {
+      if (action instanceof WebView.Preview) {
+        id = setTimeout(showPreview, 70);
```

Comment:
> I suggest a slightly longer timeout. It's still possible to see a flicker of the switcher if I don't release tab fast enough.

Can you play around with numbers and suggest one that you find to be the best, in my comparison to OSX app switcher this seemed the closest  but a lot of variables can be in a play here like the power of machine personal perception etc... I'm happy to set this to whatever you find to be the best.